### DISCUSSION TITLE
Using the atlas HCP1065 in DIPY

### DISCUSSION THREAD

**Discussion by neuroheinz** [dipy/dipy#3058] [Tractography / Tractometry]
## Description

Hello. For our work, we need to use the HCP1065 atlas by Yeh et al. We would like to clarify with you how the tractogram of the whole brain was obtained, as well as obtaining the streamline file for a specific atlas. Sincerely, Heinz A., MD.

**Comment by skoudoro** [comment]
Hi @neuroheinz, 

Thank you for your question. I convert this issue to a discussion since it is not specific to a problem in the codebase.

@BramshQamar, can you answer to @neuroheinz?

Also, you might find your answer in this documentation page: https://docs.dipy.org/stable/user_guide/data.html

**Comment by neuroheinz** [comment]
Thank you very much, @skoudoro ! I will clarify the question. I would like to know the steps that should be taken in order to get from the atlas to https://brain.labsolver.org/hcp_trk_atlas.html by Yeh et al. get an improved atlas, which is used in dipy and described in the documentation.

**Comment by BramshQamar** [comment]
Hi @neuroheinz,
Thanks for reaching out. We created a whole-brain tractogram by combining 80 bundles in the atlas together. We further divided the Corpus Colosum into three parts: Forceps Minor, Forceps Major, and Corpus Colosum middle part. We pruned some streamlines from Arcuate Fasiculus, and Uncinate Fasiculus bundles that we deemed to be spurious. 

You can download DIPY's version of the HCP atlas by Frank Yeh using the following commands:

```
from dipy.data import (fetch_bundle_atlas_hcp842, get_bundle_atlas_hcp842)
from dipy.io.streamline import load_trk
from dipy.io.utils import create_tractogram_header

atlas_file, atlas_folder = fetch_bundle_atlas_hcp842()
atlas_file, all_bundles_files = get_bundle_atlas_hcp842()

sft_atlas = load_trk(atlas_file, "same", bbox_valid_check=False)
atlas = sft_atlas.streamlines
atlas_header = create_tractogram_header(atlas_file,
                                        *sft_atlas.space_attributes)
```

You can also access the full tutorial explaining  how to fetch data and use it here: https://docs.dipy.org/stable/examples_built/segmentation/bundle_extraction.html#sphx-glr-examples-built-segmentation-bundle-extraction-py

Alternatively, you can also download data directly from [figshare](https://figshare.com/articles/dataset/Advanced_Atlas_of_80_Bundles_in_MNI_space/7375883).
Or a smaller version of the atlas containing commonly used [30 bundles](https://figshare.com/articles/dataset/Atlas_of_30_Human_Brain_Bundles_in_MNI_space/12089652).

**Reply by neuroheinz** [reply]
Thank you! But we would also like to clarify whether we can independently use the HCP1065 atlas (alternative) we need (only it currently provides standardized bundles of a number of paths)? We would also like to know if there is any specific way to get whole-brain tractograms?

**Reply by BramshQamar** [reply]
You will have to load all the bundles, concatenate them together, and save them as a .trk file. That should give you a whole-brain tractogram.

**Reply by BramshQamar** [reply]
You can download the data in .trk format [here](https://pitt-my.sharepoint.com/personal/yehfc_pitt_edu/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fyehfc%5Fpitt%5Fedu%2FDocuments%2Fpublic%2FTemplate%5FAtlas%2FHCP%5FYA1065%5Ftractography%5Fatlas%2Fpopulation%2Daveraged%2Ftracks%5Ftrk&ga=1)

**Reply by neuroheinz** [reply]
Thanks! Also, how can we get streamlines.trk which is also downloading by dipy?
